### PR TITLE
[Ready for review/merge]Rolls chainsaw surgery into prosthetic replacement, makes synthetic armblades implantable

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
tweak: Implanting chainsaws is now a prosthetic replacement instead of its own surgery.
add: You can now implant synthetic armblades (from an emagged limb grower) into people's arms to use it at its full potential.
del: Chainsaw removal surgery has been removed as well; you'll have to sever the limb and get a new one.
/:cl:

## Reason for adding
Makes the surgery code more consistent, without a snowflake surgery for chainsaws. This also gives a legitimate use for synthetic armblades, which were otherwise just a much shittier and harder to get circular saw.

### CODE REVIEW

**Comment by PKPenguin321** [comment]
Awesome :+1:

File: code/modules/surgery/amputation.dm [hunk 1]
```diff
@@ -19,5 +19,11 @@
 	var/mob/living/carbon/human/L = target
 	user.visible_message("[user] severs [L]'s [parse_zone(target_zone)]!", "<span class='notice'>You sever [L]'s [parse_zone(target_zone)].</span>")
 	if(surgery.operated_bodypart)
-		surgery.operated_bodypart.drop_limb()
+		var/obj/item/bodypart/target_limb = surgery.operated_bodypart
+		var/obj/item/held_item = L.get_item_for_held_index(target_limb.held_index)
+		target_limb.drop_limb()
+		if(held_item && held_item.flags & NODROP)
+			world << held_item
```
Comment (RemieRichards) [inline code/modules/surgery/amputation.dm]: :eyes:

**Comment by PKPenguin321** [comment]
Oh one concern though, how does this handle removing chainsaw arms?

File: code/modules/surgery/amputation.dm (hunk 1 above)
Comment (XDTM) [inline code/modules/surgery/amputation.dm]: Whoops

**Comment by XDTM** [comment]
Gotta cut the whole thing off, with amputation surgery

**Comment by PKPenguin321** [comment]
I guess that's fine considering when these were added there wasn't dismemberment so that wasn't an option

:+1:

File: code/modules/surgery/prosthetic_replacement.dm [hunk 2]
```diff
@@ -16,38 +16,58 @@
...
+		if(organ_rejection_dam)
+			target.adjustToxLoss(organ_rejection_dam)
+		user.visible_message("[user] successfully replaces [target]'s [parse_zone(target_zone)]!", "<span class='notice'>You succeed in replacing [target]'s [parse_zone(target_zone)].</span>")
+		return 1
+	else
+		target.regenerate_limb(target_zone)
+		user.visible_message("[user] finishes attaching [tool]!", "<span class='notice'>You attach [tool].</span>")
+		user.unEquip(tool)
+		qdel(tool)
+		if(istype(tool, /obj/item/weapon/twohanded/required/chainsaw))
```
Comment (KorPhaeron) [inline code/modules/surgery/prosthetic_replacement.dm]: This is pretty snowflake but I guess the rest of surgery is as well

**Pull request merged**